src/standalone/Vegetation/soil_moisture_stress.jl

```diff
@@ -0,0 +1,390 @@
+- c should be positive
+- β should be in (0,1]
+- soil_zmin should be negative
+"""
+function PiecewiseMoistureStressParameters(
+    ::Type{FT};
+    θ_high,
+    θ_low,
+    c,
+    β0,
+    soil_zmin = FT(-1.0),
+) where {FT <: AbstractFloat}
+    return PiecewiseMoistureStressParameters{FT}(
+        θ_high,
+        θ_low,
+        c,
+        β0,
+        soil_zmin,
+    )
+end
+
+Base.eltype(::PiecewiseMoistureStressParameters{FT}) where {FT} = FT
+Base.broadcastable(x::PiecewiseMoistureStressParameters) = tuple(x)
+
+"""
+    PiecewiseMoistureStressModel{FT, TMSP <: PiecewiseMoistureStressParameters{FT}}
+    <: AbstractSoilMoistureStressModel{FT}
+
+"""
+struct PiecewiseMoistureStressModel{
+    FT,
+    TMSP <: PiecewiseMoistureStressParameters{FT},
+} <: AbstractSoilMoistureStressModel{FT}
+    parameters::TMSP
+end
+
+function PiecewiseMoistureStressModel{FT}(
+    parameters::PiecewiseMoistureStressParameters{FT},
+) where {FT <: AbstractFloat}
+    return PiecewiseMoistureStressModel{eltype(parameters), typeof(parameters)}(
+        parameters,
+    )
+end
+
+ClimaLand.auxiliary_vars(model::PiecewiseMoistureStressModel) = (:βm, :ϑ_root)
+ClimaLand.auxiliary_types(model::PiecewiseMoistureStressModel{FT}) where {FT} =
+    (FT, FT)
+ClimaLand.auxiliary_domain_names(::PiecewiseMoistureStressModel) =
+    (:surface, :surface)
+
+"""
+    piecewise_moisture_stress_params_from_hydrology(
+        hydrology_cm,
+        ν,
+        θ_r;
+        c = 1.0,
+        β0 = 1.0,
+        ψ_fc = -3.36,  # Field capacity matric potential (m H2O, equivalent to -33 kPa)
+        ψ_wp = -152.9  # Wilting point matric potential (m H2O, equivalent to -1500 kPa)
+    )
+
+Helper function to create `PiecewiseMoistureStressParameters` by calculating field capacity (θ_high)
+and wilting point (θ_low) from the soil hydraulic closure model `hydrology_cm`.
+
+# Arguments
+- `hydrology_cm`: Soil hydraulic closure model (e.g., vanGenuchten or BrooksCorey)
+- `ν`: Soil porosity (m^3/m^3)
+- `θ_r`: Residual water content (m^3/m^3)
+- `soil_zmin`: Depth of the lowest soil cell face (m), must be negative
+- `c`: Curvature parameter (unitless), default 1.0
+- `β0`: Intrinsic moisture stress factor (unitless), default 1.0
+- `ψ_fc`: Field capacity matric potential (m H2O), default -3.36 m (-33 kPa)
+- `ψ_wp`: Wilting point matric potential (m H2O), default -152.9 m (-1500 kPa)
+
+The function uses the `inverse_matric_potential` function to calculate the effective
+saturation at field capacity and wilting point, then converts to volumetric water content.
+"""
+function PiecewiseMoistureStressParametersFromHydrology(
+    ::Type{FT},
+    hydrology_cm,
+    ν::Union{FT, ClimaCore.Fields.Field},
+    θ_r::Union{FT, ClimaCore.Fields.Field},
+    soil_zmin::Union{FT, ClimaCore.Fields.Field};
+    c::FT = FT(1.0),
+    β0::FT = FT(1.0),
+    ψ_fc::FT = FT(-3.36734694),    # -33 kPa, equivalent in m H2O
```

Comment:
Not sure if we need psi_fc and psi_wp if we have the retention curve params? I believe we can compute these from the retention curve.

Reply:
this is great though! This is what I mean by having a default

Reply:
ah, ok. I dont think this is a good idea (if it is known that these thresholds are not constant). I updated this so that we either use porosity and the residual as the high and low values, or we use the van genuchten curves to get the air entry and a "dry" location) based on the shape of the curve: 
<img width="1367" height="909" alt="Screenshot 2025-09-05 at 1 34 39 PM" src="https://github.com/user-attachments/assets/9bf44a53-a85c-47ea-aa5d-38a610373e88" />